src/python/pants/backend/python/tasks2/python_binary_create.py

```diff
@@ -103,6 +103,12 @@ def _create_binary(self, binary_tgt, results_dir):
 
       builder = PEXBuilder(path=tmpdir, interpreter=interpreter, pex_info=pex_info, copy=True)
 
+      if type(binary_tgt) is PythonBinary and binary_tgt.shebang:
```

Comment:
This type check has already happened, the `binary_tgt` argument came from `binaries = self.context.targets(self.is_binary)` on line 56 above.

Reply:
Valid point.